apps/web/app/api/user/stats/emails/all/get-all.ts

```diff
@@ -0,0 +1,207 @@
+    getLastEmail({ ownerEmail, direction: "newest", gmail }),
+  ]);
+  console.log("oldestEmailSaved", oldestEmailSaved);
+  console.log("newestEmailSaved", newestEmailSaved);
+
+  let after =
+    (!body.loadBefore && body.timestamp) || getTimestamp(newestEmailSaved);
+  console.log("Loading emails after:", after);
+
+  let before =
+    (body.loadBefore && body.timestamp) || getTimestamp(oldestEmailSaved);
+
+  const getMailHandler =
+    (operation: Operation) => async (timestamp: number) => {
+      let pages = 0;
+      const operationRequest = () => ({
+        ownerEmail,
+        gmail,
+        accessToken,
+        nextPageToken,
+        ...(operation === Operation.LOAD_AFTER_LATEST ||
+        operation === Operation.LOAD_AFTER_TIMESTAMP
+          ? { after: timestamp, before: undefined }
+          : { after: undefined, before: timestamp }),
+      });
+
+      while (pages < MAX_PAGES) {
+        console.log("After Page", pages);
+        let batchResponse, res;
+        try {
+          batchResponse = await saveBatch(operationRequest());
+          mailList.push(...batchResponse.emailsToPublish);
+          res = batchResponse.res;
+        } catch (error) {
+          console.log(
+            `Rate limited. Waiting ${PAUSE_AFTER_RATE_LIMIT} seconds...`,
+          );
+          await sleep(PAUSE_AFTER_RATE_LIMIT);
+          batchResponse = await saveBatch(operationRequest());
+          mailList.push(...batchResponse.emailsToPublish);
+          res = batchResponse.res;
+        }
+
+        nextPageToken = res.data.nextPageToken ?? undefined;
+
+        if (!res.data.messages || res.data.messages.length < PAGE_SIZE) break;
+        else pages++;
+      }
+      return pages;
+    };
+
+  let pages = await getMailHandler(
+    body.loadBefore
+      ? Operation.LOAD_AFTER_TIMESTAMP
+      : Operation.LOAD_AFTER_LATEST,
+  )(body.loadBefore ? before : after);
+
+  return { pages, emails: mailList };
+}
+
+async function saveBatch(
+  options: {
+    ownerEmail: string;
+    gmail: gmail_v1.Gmail;
+    accessToken: string;
+    nextPageToken?: string;
+  } & (
+    | { before: number; after: undefined }
+    | { before: undefined; after: number }
+  ),
+) {
+  const { ownerEmail, gmail, accessToken, nextPageToken, before, after } =
+    options;
+
+  // 1. find all emails since the last time we ran this function
+  const q = before
+    ? `before:${before / 1000 + 1}`
+    : after
+      ? `after:${after / 1000 - 1}`
+      : undefined;
+
+  const res = await gmail.users.messages.list({
+    userId: "me",
+    maxResults: PAGE_SIZE,
+    pageToken: nextPageToken,
+    q,
+  });
+
+  // 2. fetch each email and return them --
+  const messages = await getMessagesBatch(
+    res.data.messages?.map((m) => m.id!) || [],
+    accessToken,
+  );
+
+  const emailsToPublish: IndexedDBEmail[] = (
+    await Promise.all(
+      // why is this function async ?? This could simply be sync (including promise)
```

Comment:
good point. i fixed this in `main` now

Reply:
i'm guessing it happened because async was needed here in the past. but isn't in the current version of the code.